addons/product/models/product_template.py

```diff
@@ -183,10 +183,15 @@ def _compute_currency_id(self):
         for template in self:
             template.currency_id = template.company_id.sudo().currency_id.id or main_company.currency_id.id
 
-    @api.depends_context('company')
+    @api.depends_context('force_company')
```

Comment:
Isn't the other depends still needed (for resolving self.env.company)? Not sure how to declare that any way

Reply:
In fact, we cleaned up the multi company API, but in 13.1 (too short on time & risky to merge it in 13 just before OXP).

In 13:
* `@api.depends_context("force_company")`
Decorates compute methods depending on current company (i.e. one computed value by self.env.company or force_company context content)
* `self.with_context(force_company=cid)`
impacts ir.property, ir.sequence, the compute depending on `force_company` context and some obscure business code
* `self.with_context(allowed_company_ids=[cid])`
updates self.env.company and self.env.companies
* `self.env.company` (companies)
returns companies w.r.t user companies (`company_id` & `company_ids`) & allowed_company_ids context

In 13.1: multiple changes
* **force_company** removed
* `api.depends_context('company')` instead of `force_company`
* `self.with_company(company.id) == self.with_company(company)`
Puts given company as main company in the context (self.env.company)
* `self.env.company(companies)`
Considers sudo, may raise if invalid `allowed_company_ids` context

A [multi-company guidelines Documentation](https://www.odoo.com/documentation/13.0/howtos/company.html) is available (and up to date in 13 & 13.1) 

The current `@api.depends_context('company')` is wrong in 13 (I did backport it too much), but correct in 13.1 with the new API.

Reply:
Thanks for the explanation! I will check the documentation

Reply:
Do not hesitate to contact me if sthg is unclear on the page (in 13 or master), I tried to explain the different multi-company tools/behaviors clearly, but some things might be unclear/missing ;).